win32ss/gdi/dib/dib16bpp.c

```diff
@@ -353,42 +669,84 @@ DIB_16BPP_BitBltSrcCopy(PBLTINFO BltInfo)
         *((WORD *)DestBits) = (WORD)XLATEOBJ_iXlate(
           BltInfo->XlateSourceToDest, xColor);
 
-        SourceBits += 3;
+        if (bLeftToRight)
+        {
+          SourceBits -= 3;
+        }
+        else
+        {
+          SourceBits += 3;
+        }
         DestBits += 2;
       }
-      SourceLine += BltInfo->SourceSurface->lDelta;
+      if (bTopToBottom)
+      {
+        SourceLine -= BltInfo->SourceSurface->lDelta;
+      }
+      else
+      {
+        SourceLine += BltInfo->SourceSurface->lDelta;
+      }
```

Comment:
and this in one single line too:
```c
    INCREMENT(SourceLine, bTopToBottom, BltInfo->SourceSurface->lDelta);
```

Reply:
@HBelusca, How about a name of "DecOrInc" instead of "DECREMENT" so that the macro could look this this:

    #define DecOrInc(var, decTrue, amount) \
        (var = decTrue ? (var - amount) : (var + amount))

Reply:
Dunno.... The "natural" (this is subjective) operation is incrementation (and that's what was present in the original code), so something like ADVANCE or INCREMENT, with the direction TRUE to actually increment and FALSE to decrement, like:
```c
#define ADVANCE(var, direction, amount) \
    ((var) = (direction) ? ((var) + (amount)) : ((var) - (amount)))
```
(note: code style requires macro names in all caps, and the argument variables within the macro definition to be within parentheses).

But then that code example would become:
```c
ADVANCE(SourceLine, !bTopToBottom, BltInfo->SourceSurface->lDelta);
```
since we actually want to decrease `SourceLine` when `bTopToBottom` is TRUE (or increase otherwise).

Reply:
@HBelusca, Thank you for the help with the standards. I am not really familiar with them. How about this:

    #define DEC_OR_INC(var, decTrue, amount) \
        ((var) = (decTrue) ? ((var) - (amount)) : ((var) + (amount)))

Reply:
Fixed by changing this code the new DEC_OR_INC macro.